packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -199,9 +199,9 @@ export class Pipeline extends PipelineBase {
 
     validateName('Pipeline', this.physicalName);
 
-    // only one of artifactBucket and crossRegionReplicationBuckets can be supplied
-    if (props.artifactBucket && props.crossRegionReplicationBuckets) {
-      throw new Error('Only one of artifactBucket and crossRegionReplicationBuckets can be specified!');
+    // only one of artifactBucket and artifactBuckets can be supplied
+    if (props.artifactBucket && props.artifactBuckets) {
```

Comment:
So now we have artifactBucket and artifactBuckets?? That sounds awfully confusing... why would I need to use one and not the other? Maybe replicationBuckets?

Reply:
> why would I need to use one and not the other?

You would use `artifactBuckets` if you had a cross-region CodePipeline, `artifactBucket` otherwise.

> Maybe replicationBuckets?

The issue is that you can also specify the Pipeline's Bucket using this property (when giving a Bucket for the region the pipeline itself is in), so the name `replicationBuckets` is, strictly speaking, not precise enough.

Reply:
What’s the benefit of specifying the pipeline’s bucket here and not in artifactsBucket? 

I still feel replicationBuckets is a better name, and it’s ok if replication “succeeds” with no-op in the case of same-region.

Reply:
> What’s the benefit of specifying the pipeline’s bucket here and not in artifactsBucket?

Because you can't specify both `artifactBucket` and `artifactBuckets` property (it's a validation error, it results in an exception being thrown).

Reply:
Can we smooth this out in the construct? If users specify both and replicationBuckets don’t have a bucket for the current environment, you can simply merge them when you set the L1

Reply:
The behavior of making `artifactBucket` and `artifactBuckets` exclusive with each other was asked by the Pipeline team, so I wouldn't want to remove it right now.